backend/s3/s3.go

```diff
@@ -2661,7 +2667,14 @@ func (o *Object) Remote() string {
 	return o.remote
 }
 
-var matchMd5 = regexp.MustCompile(`^[0-9a-f]{32}$`)
+// Remote returns the match md5 regexp
+func (o *Object) matchMd5() *regexp.Regexp {
+	matchMd5Str := "^[0-9a-f]{32}$"
+	if o.fs.opt.UseMultipartEtagAsMd5 {
+		matchMd5Str = "^[0-9a-f]{32}(-[0-9]+)?$"
+	}
```

Comment:
Please make both regexp's statically compiled outside the function.
Name them e.g. `matchNormalMd5` and `matchMultipartMd5`.
The will return either one depending on the flag without compiling it every time.
Don't needelessly load golang regexp compiler/cache.